### PR TITLE
disable future selection for date of death field

### PR DESCRIPTION
**Pull request by GangulaMounika**

### CODE REVIEW

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]
Using the 'minDate' of the 'DatePicker' does prevent picking future dates from the calendar pop up however, there are inconsistencies in how it handles manual entry.  In add patient the is a 'Please enter valid date' message that is preventing the other validations from occurring.  It's also not aligned with the designs.

We may not be able to use this functionality of the `DatePicker` and have to enforce this via validations.  

There are many validations handled with the [rules from React Hook Form](https://react-hook-form.com/docs/useform/register#options).  There is an example of a custom validation function being used in the [PhoneEmailEntryForm](https://github.com/CDCgov/NEDSS-Modernization/blob/68b62e1cfed064d1140fc03f57b5d5f9cb5cea39/apps/modernization-ui/src/pages/patient/profile/phoneEmail/PhoneEmailEntryForm.tsx#L134)

File: apps/modernization-ui/src/components/FormInputs/DatePickerInput.tsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ type DatePickerProps = {
     flexBox?: boolean;
     required?: boolean;
     disabled?: boolean;
+    maxDate?: string;
```
[the remark this answers is not in the document]
Comment (GangulaMounika) [reply apps/modernization-ui/src/components/FormInputs/DatePickerInput.tsx:22]: Unfortunately this maxDate that Datepicker component of truss work takes a string.
I put this as a conditional as not always our custom "DatepickerInput" might need this functionlity.

File: apps/modernization-ui/src/pages/addPatient/components/otherInfoFields/OtherInfoFields.tsx [hunk 2]
```diff
@@ -22,6 +22,13 @@ export default function OtherInfoFields({ id, title, control, coded }: Props) {
     const currentBirthday = useWatch({ control, name: 'dob' });
     const age = useMemo(() => calculateAge(currentBirthday), [currentBirthday]);
 
+    const getCurrentLocalDate = () => {
```
[the remark this answers is not in the document]
Comment (GangulaMounika) [reply apps/modernization-ui/src/pages/addPatient/components/otherInfoFields/OtherInfoFields.tsx:25]: I agree too, but not all consumers of DatePickerInput might need the same date as the max date. 
We could enhance it at the time such a requirement comes though and go with your suggestion for now. 
Do you think maxDate should be converted into a boolean then and reamed as showMaxDate ?

[the remark this answers is not in the document]
Comment (adamloup-enquizit) [reply apps/modernization-ui/src/pages/addPatient/components/otherInfoFields/OtherInfoFields.tsx:25]: I was thinking that our `DatePickerInput` component could take the maxDate as a `Date` object and internally convert it into the `string` format required by `DatePicker` if it's present.

**Review by adamloup-enquizit — DISMISSED** [review]

**Review by adamloup-enquizit — APPROVED** [review]

**Pull request merged**